From 0b45a3e30d59ae84d483be731785aeb09e9ab83d Mon Sep 17 00:00:00 2001
From: Alexander Forsch <alexander.forsch@bytesatwork.ch>
Date: Fri, 19 Jun 2026 16:35:47 +0200
Subject: [PATCH] arm64: dts: ti: k3-am625-bytedevkit: Adapt OLDI port
 definitions

The OLDI port layout changed in Linux 6.18. The oldi0_ports and
oldi1_ports labels were removed and replaced by individual port nodes
(oldi0_port0/1 and oldi1_port0/1).

Update the Bytedevkit device trees to use the new OLDI port structure
and endpoint definitions so they compile with Linux 6.18.

Signed-off-by: Alexander Forsch <alexander.forsch@bytesatwork.ch>
---
 .../boot/dts/ti/k3-am625-bytedevkit-2-2.dts   | 42 +++++++------------
 .../arm64/boot/dts/ti/k3-am625-bytedevkit.dts | 42 +++++++------------
 2 files changed, 28 insertions(+), 56 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am625-bytedevkit-2-2.dts b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit-2-2.dts
index 169c0669f4fb..62a14591981b 100644
--- a/arch/arm64/boot/dts/ti/k3-am625-bytedevkit-2-2.dts
+++ b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit-2-2.dts
@@ -262,41 +262,27 @@ &oldi1 {
 	status = "okay";
 };
 
-&oldi0_ports {
-	#address-cells = <1>;
-	#size-cells = <0>;
-
-	port@0 {
-		reg = <0>;
-		oldi_0_in: endpoint {
-			remote-endpoint = <&dpi0_out0>;
-		};
+&oldi0_port0 {
+	oldi_0_in: endpoint {
+		remote-endpoint = <&dpi0_out0>;
 	};
+};
 
-	port@1 {
-		reg = <1>;
-		oldi_0_out: endpoint {
-			remote-endpoint = <&lcd_in0>;
-		};
+&oldi0_port1 {
+	oldi_0_out: endpoint {
+		remote-endpoint = <&lcd_in0>;
 	};
 };
 
-&oldi1_ports {
-	#address-cells = <1>;
-	#size-cells = <0>;
-
-	port@0 {
-		reg = <0>;
-		oldi_1_in: endpoint {
-			remote-endpoint = <&dpi0_out1>;
-		};
+&oldi1_port0 {
+	oldi_1_in: endpoint {
+		remote-endpoint = <&dpi0_out1>;
 	};
+};
 
-	port@1 {
-		reg = <1>;
-		oldi_1_out: endpoint {
-			remote-endpoint = <&lcd_in1>;
-		};
+&oldi1_port1 {
+	oldi_1_out: endpoint {
+		remote-endpoint = <&lcd_in1>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts
index 23318a90b7c6..0f6654bdd491 100644
--- a/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts
+++ b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts
@@ -250,41 +250,27 @@ &oldi1 {
 	status = "okay";
 };
 
-&oldi0_ports {
-	#address-cells = <1>;
-	#size-cells = <0>;
-
-	port@0 {
-		reg = <0>;
-		oldi_0_in: endpoint {
-			remote-endpoint = <&dpi0_out0>;
-		};
+&oldi0_port0 {
+	oldi_0_in: endpoint {
+		remote-endpoint = <&dpi0_out0>;
 	};
+};
 
-	port@1 {
-		reg = <1>;
-		oldi_0_out: endpoint {
-			remote-endpoint = <&lcd_in0>;
-		};
+&oldi0_port1 {
+	oldi_0_out: endpoint {
+		remote-endpoint = <&lcd_in0>;
 	};
 };
 
-&oldi1_ports {
-	#address-cells = <1>;
-	#size-cells = <0>;
-
-	port@0 {
-		reg = <0>;
-		oldi_1_in: endpoint {
-			remote-endpoint = <&dpi0_out1>;
-		};
+&oldi1_port0 {
+	oldi_1_in: endpoint {
+		remote-endpoint = <&dpi0_out1>;
 	};
+};
 
-	port@1 {
-		reg = <1>;
-		oldi_1_out: endpoint {
-			remote-endpoint = <&lcd_in1>;
-		};
+&oldi1_port1 {
+	oldi_1_out: endpoint {
+		remote-endpoint = <&lcd_in1>;
 	};
 };
 
